src/fromPairsMap.ts

```diff
@@ -0,0 +1,120 @@
+import _ from 'lodash'
+
+export default fromPairsMap
+
+/**
+ * Applies `fromPairs` to the result of mapping given `iteratee` to given
+ * collection.
+ *
+ * @param collection Collection to iterate over
+ * @param iteratee Function invoked per iteration
+ * @returns New object
+ */
+export function fromPairsMap<T, PropertyName, TResult> (
+  collection: T[] | null | undefined,
```

Comment:
Why is the definition different here than in other? where we have 
```
CollectionItem extends any,
  Collection extends ArrayLike<CollectionItem | null | undefined>,
  PropertyName,
  TResult
> (
collection: Collection,
iteratee: ObjectIterator<CollectionItem, [PropertyName, TResult] | null | undefined>
```
we should have consistency.

Reply:
Because the definition of `_.map` forces us to be more strict. If we use a more generic definition then the compiler wrongly matches our `_.map` usage to the one of using an `Object` as iteratee, which always returns an array of boolean (`boolean[]`) which is not allowed by `_.fromPairs`.

It's tricky because Lodash types are really complex but I think this is good enough (see also [this comment](https://github.com/geoblink/lodash-mixins/pull/11#discussion_r257683000) about why types of those methods are not really so important)